services/wallet/thirdparty/coingecko/client_test.go

```diff
@@ -0,0 +1,98 @@
+package coingecko
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func setupTest(t *testing.T, delay time.Duration, response []byte) (*httptest.Server, func()) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(delay * time.Second)
```

Comment:
I think `delaySecs` should be `int` if you want to use it this way. The naming means a  count of seconds, but I see here `time.Duration`, which is unit-of-time agnostic.

The best is to rename `delaySecs` to `delay` and keep the `time.Duration`, pass it as such and later on, where you use it, pass `1 * time.Second`